src/Listener.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * File: Listener.h
+ * Author: Craig Cogdill
+ * Created: August 14, 2015 1:08PM
+ */
+
+#pragma once
+#include <vector>
+#include <string>
+#include <memory>
+
+class Alien;
+class Rifle;
+
+class Listener {
+   public:
+      Listener() = delete;
+      Listener(const std::string& notificationQueue, const std::string& handshakeQueue, const std::string& program);
```

Comment:
I do not like the double construction phase. 

1) Constructor + 2) Initialize() before it is usable. This, multi-step initialization,  is an anti-pattern IMHO.

I'd rather have a private/protected constructor and a factory method that setups everything and returns the Listener in a unique ptr. 

I.e. exactly what your "CreateFeedbackShooter" is doing.  If you need help to get this done I can sit down with you.